float64 ball_distance
float64 ball_center_x
